File: typescript/word-count/word-count.ts
export default class Words {
  count(input: string): Map<string, number> {
    const result = new Map<string, number>()
    input.toLowerCase().split(/ |\n|\t/).forEach((word) => {
      if (word.length > 0) {
        const value = result.get(word) || 0
        result.set(word, value + 1)
      }
    })
    return result
  }
}

File: clojure/word-count/src/word_count.clj
(ns word-count
  (:require [clojure.string :as string]))

(defn counter [[key value]]
  [key (count value)])

(defn word-count [input]
  (let [words (-> input
                  string/lower-case
                  (string/replace #":|,|!|&|@|\$|%|\^|&|\"" "")
                  (string/split #"\s+"))]
    (->> words
         (group-by identity)
         (map counter)
         (into {}))))
